### ISSUE TITLE
DEPRECATION WARNING: Relation#all is deprecated.

### ISSUE THREAD

**Issue by danielgatis** [activeadmin/inherited_resources#290]
DEPRECATION WARNING: Relation#all is deprecated. If you want to eager-load a relation, you can call #load (e.g. `Post.where(published: true).load`). If you want to get an array of records from a relation, you can call #to_a (e.g. `Post.where(published: true).to_a`).

(called from collection at ... /lib/inherited_resources/base_helpers.rb:26)

**Comment by virtualfunction** [comment]
Probably not the nicest work around but in base_helpers.rb I amended the collection method to:

``` ruby
def collection
  get_collection_ivar || begin
    chain = end_of_association_chain
    use_scoped = chain.respond_to?(:scoped) && !ActiveRecord.const_defined?(:DeprecatedFinders)
    items = case
      when use_scoped then chain.scoped
      when chain.is_a?(ActiveRecord::Relation) then chain
      else chain.all
    end
    set_collection_ivar items
  end
end
```

**Comment by dim** [comment]
I tried to submit a pull request for rails, but it was rejected: https://github.com/rails/rails/issues/11310. Seems like this needs to be fixed in inherited_resources

**Comment by joelmoss** [comment]
Resolved in #298
